README.md

```diff
@@ -4,9 +4,113 @@
+|Project|Build Hook|Build Hook|TheCodingMachine\GitlabHook\Model\Build|
+|Group|Project created|System Hook|TheCodingMachine\GitlabHook\Model\ProjectCreate|
+|Group|Project destroyed|System Hook|TheCodingMachine\GitlabHook\Model\ProjectDestroy|
+|Group|Project renamed|System Hook|TheCodingMachine\GitlabHook\Model\ProjectRename|
+|Group|Project transferred|System Hook|TheCodingMachine\GitlabHook\Model\ProjectTrasnfer|
+|Group|New Team Member|System Hook|TheCodingMachine\GitlabHook\Model\TeamMemberAdd|
+|Group|Team Member Removed|System Hook|TheCodingMachine\GitlabHook\Model\TeamMemberRemove|
+|Group|User created|System Hook|TheCodingMachine\GitlabHook\Model\UserCreate|
+|Group|User removed|System Hook|TheCodingMachine\GitlabHook\Model\UserDestroy|
+|Group|User failed login|System Hook|TheCodingMachine\GitlabHook\Model\UserFailedLogin|
+|Group|User renamed|System Hook|TheCodingMachine\GitlabHook\Model\UserRename|
+|Group|Key added|System Hook|TheCodingMachine\GitlabHook\Model\KeyCreate|
+|Group|Key removed|System Hook|TheCodingMachine\GitlabHook\Model\KeyDestroy|
+|Group|Group created|System Hook|TheCodingMachine\GitlabHook\Model\GroupCreate|
+|Group|Group removed|System Hook|TheCodingMachine\GitlabHook\Model\GroupDestroy|
+|Group|Group renamed|System Hook|TheCodingMachine\GitlabHook\Model\GroupRename|
+|Group|New Group Member|System Hook|TheCodingMachine\GitlabHook\Model\UserGroupAdd|
+|Group|Group Member Removed|System Hook|TheCodingMachine\GitlabHook\Model\UserGroupRemove|
+|Group|Push events|System Hook|TheCodingMachine\GitlabHook\Model\Push|
+|Group|Tag events|System Hook|TheCodingMachine\GitlabHook\Model\Tag|
+|Group|Merge request events|System Hook|TheCodingMachine\GitlabHook\Model\MergeRequest|
+|Group|Repository Update events|System Hook|TheCodingMachine\GitlabHook\Model\RepositoryUpdate|
+
+
+Example
+-------
+
+```php
+<?php
+namespace Test;
+
+class Listener implements HookListenerInterface {
+    /**
+     * @param \TheCodingMachine\GitlabHook\EventInterface $event
+     */
+     public function onEvent(EventInterface $event) {
+         // Compute Push event
+         if($event instanceof TheCodingMachine\GitlabHook\Model\Push::class) {
+             // Display before
+             echo $event->getBefore();
+             // Display the project name
+             echo $event->getProject()->getName();
+         }
+         // Compute MergeRequest event
+         if($event instanceof TheCodingMachine\GitlabHook\Model\MergeRequest::class) {
+             // Display target branch (this is in object_attributes)
+             echo $event->getTargetBranch();
+             // Get initial payload
+             var_dump($event->getPayload());
+         }
+     }
+}
+?>
+``` 
+
+```php
+
+// You must inject this middleware in the middleware pipe of your favorite framework.
+// See your framework documentation on how to do that.
+// Create your listener
+$listener = new Test/Listerner();
+// Add it to the hook computer
```

Comment:
`// Add it to the hook computer` => `// Register your listener in the main HookReceiver instance`